gtk/src/toga_gtk/widgets/canvas.py

```diff
@@ -35,9 +38,48 @@ def create(self):
                 | Gdk.EventMask.BUTTON_MOTION_MASK
             )
         else:  # pragma: no-cover-if-gtk3
-            pass
+            self.native.set_draw_func(WeakrefCallable(self.gtk_draw_callback))
+            self.native.connect("resize", self.gtk_resize)
+
+            self.gesture_click = {}
+            self.gesture_drag = {}
+            for button in (1, 3):
+                self.gesture_click[button] = Gtk.GestureClick()
+                self.native.add_controller(self.gesture_click[button])
+                self.gesture_click[button].set_button(button)
+                self.gesture_click[button].connect(
+                    "pressed", WeakrefCallable(self.mouse_down)
+                )
+                self.gesture_click[button].connect(
+                    "released", WeakrefCallable(self.mouse_up)
+                )
+
+                self.gesture_drag[button] = Gtk.GestureDrag()
+                self.native.add_controller(self.gesture_drag[button])
+                self.gesture_drag[button].set_button(button)
+                self.gesture_drag[button].connect(
+                    "drag-update", WeakrefCallable(self.mouse_drag)
```

Comment:
This isn't entirely a problem of your making, but we should clean it up since we're in the area - GTK handlers should all be prefixed with `gtk_`, and follow the naming of the underlying event (So, `gtk_drag_update` in this case; similarly for mouse press and release events)